File: EjercicioGenerecidad5/src/main/java/com/mycompany/ejerciciogenerecidad5/EjercicioGenerecidad5.py
from typing import Generic, TypeVar, List

T = TypeVar('T')

class Pila(Generic[T]):
    def __init__(self):
        self.elementos: List[T] = []

    def apilar(self, elemento: T):
        self.elementos.append(elemento)

    def desapilar(self):
        if self.elementos:
            return self.elementos.pop()
        return None

    def mostrar(self):
        print("Contenido de la pila:")
        for e in reversed(self.elementos):
            print(e)

    def esta_vacia(self):
        return len(self.elementos) == 0
from EjercicioGenerecidad5 import Pila

pila_int = Pila[int]()
pila_int.apilar(1)
pila_int.apilar(2)
pila_int.apilar(3)
pila_int.mostrar()
print("Desapilado:", pila_int.desapilar())
pila_int.mostrar()

pila_str = Pila[str]()
pila_str.apilar("A")
pila_str.apilar("B")
pila_str.apilar("C")
pila_str.mostrar()
print("Desapilado:", pila_str.desapilar())
pila_str.mostrar()

File: EjercicioGenerecidad5/src/main/java/com/mycompany/ejerciciogenerecidad5/EjercicioGenerecidad5.java
package com.mycompany.ejerciciogenerecidad5;
import java.util.*;
/**
 *
 * @author Vladimir Ohara
 */
class Pila<T> {
    private List<T> elementos;

    public Pila() {
        elementos = new ArrayList<>();
    }
//Elinciso A 
    public void apilar(T elemento) {
        elementos.add(elemento);
    }
//EL inciso B
    public T desapilar() {
        if (!elementos.isEmpty()) {
            return elementos.remove(elementos.size() - 1);
        }
        return null;
    }

    public void mostrar() {
        System.out.println("Contenido de la pila:");
        for (int i = elementos.size() - 1; i >= 0; i--) {
            System.out.println(elementos.get(i));
        }
    }

    public boolean estaVacia() {
        return elementos.isEmpty();
    }
}
public class EjercicioGenerecidad5 {

    public static void main(String[] args) {
        // inciso C
        Pila<Integer> pilaEnteros = new Pila<>();
        pilaEnteros.apilar(10);
        pilaEnteros.apilar(20);
        pilaEnteros.apilar(30);

        Pila<String> pilaStrings = new Pila<>();
        pilaStrings.apilar("uno");
        pilaStrings.apilar("dos");
        pilaStrings.apilar("tres");
        //El inciso D
        pilaEnteros.mostrar();
        System.out.println("Desapilado: " + pilaEnteros.desapilar());
        pilaEnteros.mostrar();
        
        pilaStrings.mostrar();
        System.out.println("Desapilado: " + pilaStrings.desapilar());
        pilaStrings.mostrar();
    }
}
